From patches Mon Sep 17 00:00:00 2001
From: liguang <lig.fnst@cn.fujitsu.com>
To: Igor Mammedov <imammedo@redhat.com>, liguang <lig.fnst@cn.fujitsu.com>,
 afaerber@suse.de, li guang <lig.fnst@cn.fujitsu.com>, qemu-devel@nongnu.org
Cc: qemu-devel@nongnu.org, afaerber@suse.de, liguang <lig.fnst@cn.fujitsu.com>
Date: Wed, 10 Apr 2013 16:02:28 +0800
Message-id: 1365580949-32467-1-git-send-email-lig.fnst@cn.fujitsu.com
Subject: [PATCH 1/2] target-i386: remove cpu reset callback

we should remove it now, if not, cpu reset
will do 2 times.

Signed-off-by: liguang <lig.fnst@cn.fujitsu.com>
---
 target-i386/cpu.c |    9 ---------
 1 files changed, 0 insertions(+), 9 deletions(-)

diff --git a/target-i386/cpu.c b/target-i386/cpu.c
index 356378c..6e13053 100644
--- a/target-i386/cpu.c
+++ b/target-i386/cpu.c
@@ -2024,13 +2024,6 @@ bool cpu_is_bsp(X86CPU *cpu)
 {
     return cpu_get_apic_base(cpu->env.apic_state) & MSR_IA32_APICBASE_BSP;
 }
-
-/* TODO: remove me, when reset over QOM tree is implemented */
-static void x86_cpu_machine_reset_cb(void *opaque)
-{
-    X86CPU *cpu = opaque;
-    cpu_reset(CPU(cpu));
-}
 #endif
 
 static void mce_init(X86CPU *cpu)
@@ -2140,8 +2133,6 @@ static void x86_cpu_realizefn(DeviceState *dev, Error **errp)
     }
 
 #ifndef CONFIG_USER_ONLY
-    qemu_register_reset(x86_cpu_machine_reset_cb, cpu);
-
     if (cpu->env.cpuid_features & CPUID_APIC || smp_cpus > 1) {
         x86_cpu_apic_init(cpu, &local_err);
         if (local_err != NULL) {
-- 
1.7.2.5


